components/brave_wallet_ui/common/wallet_api_proxy.ts

```diff
@@ -88,6 +88,22 @@ export class WalletApiProxy {
   addBraveWalletServiceObserver (store: Store) {
     const braveWalletServiceObserverReceiver = new BraveWallet.BraveWalletServiceObserverReceiver({
       onActiveOriginChanged: function (originInfo) {
+        const state = store.getState().wallet
+
+        // check that the origin has changed from the stored values
+        // in any way before dispatching the update action
+        if (
+          state.activeOrigin.eTldPlusOne === originInfo.eTldPlusOne &&
+          state.activeOrigin.origin.host === originInfo.origin.host &&
+          state.activeOrigin.origin.nonceIfOpaque?.high === originInfo.origin.nonceIfOpaque?.high &&
+          state.activeOrigin.origin.nonceIfOpaque?.low === originInfo.origin.nonceIfOpaque?.low &&
+          state.activeOrigin.origin.port === originInfo.origin.port &&
+          state.activeOrigin.origin.scheme === originInfo.origin.scheme &&
+          state.activeOrigin.originSpec === originInfo.originSpec
```

Comment:
Is there a better way to do this? We'll most certainly forget to update this if ever there's a new field introduced to `OriginInfo`. Maybe have a a generic `isEqual(x, y)` util. Found one on SO: https://stackoverflow.com/a/16788517/1946230

Reply:
Added util here: https://github.com/brave/brave-core/pull/14833/commits/6ac56dfbeab85a1dd0f9fcb1ea6cd8e01b6404e2

Applied util here: https://github.com/brave/brave-core/pull/14833/commits/8be5cd42c39f7cf9838050e81acc035435ea6429